decidim-elections/app/commands/decidim/votings/census/admin/increment_dataset_processed_rows.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Votings
+    module Census
+      module Admin
+        # A command with the business logic to create increment the dataset
+        # processed rows and change the state when the last is processed
+        class IncrementDatasetProcessedRows < Rectify::Command
+          def initialize(dataset)
+            @dataset = dataset
+          end
+
+          # Executes the command. Broadcast this events:
+          # - :ok when everything is valid
+          # - :invalid when the form wasn't valid and couldn't proceed-
+          #
+          # Returns nothing.
+          def call
+            return broadcast(:invalid) unless dataset
+
+            # rubocop:disable Rails/SkipsModelValidations
+            Dataset.increment_counter(:csv_row_processed_count, dataset.id)
```

Comment:
⚠️ I have two thoughts related to this:
* Even when I think this method is better, this could be faster if you do it directly in the CreateDatum command. I don't know if running a new command for each csv row could be a performance issue.
* Is this operation atomic? If we have several jobs running the same operation we could have race conditions problems. I saw that this is run in the `after_perform` part of the job, but I don't know exactly how it works.

Reply:
First this was in the `CreateDatum` command but as it updates the `Dataset` model extracting it here seemed more appropriate.
In the future this command could be used to send an email when `all_rows_processed?`.

On the other hand, I used the class method `Dataset.increment_counter` instead of the instance method `dataset.increment` for that reason according [this post](https://til.icelab.com.au/manually-incrementing-count-columns/).